contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -398,42 +400,83 @@ contract InterchainTokenService is
         );
     }
 
+    // Returns the amount of token that this call is worth. If `tokenAddress` is `0`, then value is in terms of the native token, otherwise it's in terms of the token address.
+    function contractCallValue(
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) public view virtual onlyRemoteService(sourceChain, sourceAddress) notPaused returns (address, uint256) {
+        (, bytes32 tokenId, , uint256 amount) = abi.decode(payload, (uint256, bytes32, bytes, uint256));
+
+        ITokenManager tokenManager = ITokenManager(getValidTokenManagerAddress(tokenId));
+        return (tokenManager.tokenAddress(), amount);
+    }
+
+    function expressExecute(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceService,
+        bytes calldata payload
+    ) external payable notPaused {
```

Comment:
we can keep the paused check for now since this can also do arbitrary contract execution